janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/JanusGraphStep.java

```diff
@@ -78,7 +80,14 @@ public String toString() {
 
     @Override
     public void addAll(Iterable<HasContainer> has) {
-        Iterables.addAll(hasContainers, has);
+        has.forEach(hasContainer -> {
```

Comment:
Yep, the last update I pushed resolves this by moving the common code to a static HasStepFolder method that's then called from JanusGraphStep and JanusGraphVertexStep (see [updated diff](https://github.com/JanusGraph/janusgraph/pull/208/files)). Good call on that (as usual) by the way.